### PR TITLE
Audit & fix `algorithms.md` page

### LINKED ISSUE #28: Audit & fix `algorithms.md` page
This is a sub-issue of https://github.com/Seneca-ICTOER/IPC144/issues/18

## Check List

- [x] Look for any typos
- [x] Make sure all Markdown is correctly done. The conversion was done automatically by a tool, so there may be more idiomatic ways to do some things
- [x] Check the page in a Desktop browser. Does it look OK? Are there any issues that need to be fixed?
- [x] Check the page in a Mobile browser. Does it look OK? Are there any issues that need to be fixed?
- [x] Try the page in both Light and Dark mode. Are there any issues that need to be fixed?
- [x] Try running the page through Lighthouse: https://developers.google.com/web/tools/lighthouse. Are there any issues that need to be fixed?
- [x] Try running the page through Web Hint: https://webhint.io/. Are there any issues that need to be fixed?
- [x] Tables that are really wide (types.md) could be converted to use SVG or flex/grid. Make wide visual tables fit in the available viewport. Another example is the table in docs/D-Modularity/pointers.md, where we should reduce the vertical padding/margins on tables. The tables are too large, and probably shouldn't be tables at all.
- [x] Prefer Markdown tables vs. raw HTML tags
- [x] Fix colours in tables so it uses global styles vs. inline styles
- [x] Use alt tags for all images. See https://supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions
- [x] Avoid blockquote for indentation, and prefer q or blockquote only when actually quoting. Use Admonitions instead. Example: `docs/D-Modulatirty/pointers.md`: there are places where > Blockquote are being used and Admonitions could replace it. “Note:” Lightbulb info Admonition.
- [x] Make sure images work in dark and light modes. For example, `docs/8-Computations/logic.md` Flags image is white background, needs to work in dark. Create a “negative” or SVG so we can do Dark Mode? Can we programmatically create this negative?
- [x] Change colours for tables (don’t use Yellow). We should use the default colour theme.
- [x] Avoid using `<b>` and `<i>` to do emphasis.
- [x] Use backticks and all code for memory addresses and such in tables that use things like void, int, function names, memory addresses.
- [x] Make sure all code blocks use syntax highlighting
- [x] Centre images horizontally (esp. when they are small) so they work well at all responsive sizes
- [x] Check sub-script and super-script that they are correctly used in all cases
- [x] Consider including inter-site links to other pages that have related info
- [x] All links should have descriptive text for accessibility
- [x] Make sure that sub-headings (right-hand side) for each document are correct.
- [x] Improve readability of code comments (i.e., `//` and `/*...*/`) in dark mode (even light mode might need some work)
- [x] In places where `***...***` was used as a fourth heading level, use `####` instead.
- [x] Fix Frontmatter for page to include proper id, title, slug, etc.

### PR DESCRIPTION
**Pull request by oliver-pham**
Fix #28 

- Fix Frontmatter for page to include `id`, `title`, `slug`, and `description`
- Replace `**...**` used as a fourth heading level with `####`
- Fix all code to be wrapped with backticks
- Replace blockquote with Admonitions for clarity
- Include inter-site links to relevant pages
- Fix typo

### CODE REVIEW

**Review by minhhang107 — CHANGES_REQUESTED** [review]
- Line 25 should use `find`
```
Search algorithms finds the index of ...
```
- In `find()` (line 42) and `main()` (line 57): consider removing extra indentation to maintain consistency
```c
int find(int skuData[], int skuCount, int findSKU)
{
        int i, skuIndex = -1;
        for (i = 0; skuIndex < 0 && i < skuCount; i++)
        ...
        return skuIndex;
}
```

File: docs/F-Refinements/algorithms.md [hunk 1]
```diff
@@ -137,14 +144,17 @@ SKU   : 4633
...
+Masking algorithms distinguish certain array elements from all other elements.  The masking array is a [parallel array](/C-Data-Structures/arrays#parallel-arrays "Parallel Arrays") with respect to the other arrays in the set.  The elements of the masking array are [flags](/B-Computations/logic#flags "Flags") that identify inclusion or exclusion. 
 
-Consider the following program, which calculates the total purchase price for a set of products.  The user enters the sku for each product purchased and the quantity.  Some products attract HST (Harmonized Sales Tax), while others do not.  We store the skus, unit prices and tax status in three parallel arrays.  The tax status array is the masking array.  The user enters 0 to terminate input.
+Consider the following program, which calculates the total purchase price for a set of products.  The user enters the SKU (Stock-Keeping Unit) for each product purchased and the quantity.  Some products attract HST (Harmonized Sales Tax), while others do not.  We store the skus, unit prices and tax status in three parallel arrays.  The tax status array is the masking array.  The user enters 0 to terminate input.
```
Comment (lyu4321) [inline docs/F-Refinements/algorithms.md:157]:
```suggestion
Consider the following program, which calculates the total purchase price for a set of products.  The user enters the SKU (Stock-Keeping Unit) for each product purchased and the quantity.  Some products attract HST (Harmonized Sales Tax), while others do not.  We store the SKUs, unit prices and tax status in three parallel arrays.  The tax status array is the masking array.  The user enters 0 to terminate input.
```

Comment (lyu4321) [reply docs/F-Refinements/algorithms.md:157]: Also in line 459, `swap` should be `swaps` such as 

```To simulate mixing, the algorithm picks the index of a reference element, randomly picks the index of another element further along in the array and swaps the values stored in the two elements.```

**Comment by lyu4321** [comment]
Also in line 459, `swap` should be `swaps` such as:

```To simulate mixing, the algorithm picks the index of a reference element, randomly picks the index of another element further along in the array and swaps the values stored in the two elements.```

**Comment by CameronGray1210** [comment]
Thanks - good work!

**Pull request merged**